distributed/scheduler.py

```diff
@@ -2336,6 +2378,7 @@ def decide_worker(self, ts: TaskState) -> WorkerState:
                 partial(self.worker_objective, ts),
             )
         else:
+            # Fastpath when there are no related tasks or restrictions
```

Comment:
I'm realizing that this codepath will now only be rarely triggered (when there are 0 deps, but also the TaskGroup is small). Do we need to add this round-robining into our selection of a new worker for root-ish tasks? (Since we know we'll be running the tasks on every worker, I'm not sure it matters much that we may always start with the same one in an idle cluster.)

Reply:
Possibly.  More broadly this is probably a good reminder that while we run this on some larger example computations we should also remember to try looking at some profiles of the scheduler to see if/how things have changed.

Reply:
@gjoseph92 did you run into issues with this yet?  I'm curious, have you tried using many workers?  (for some sensible definition of many)

Reply:
@gjoseph92 did this come up in profiling?  this seems like the only pending comment.  I'd like to get this in if possible

Reply:
It did not come up in profiling, and I haven't run into any issues with it. I feel pretty confident that round-robining is irrelevant when we're running TaskGroups larger than the cluster. I mostly brought it up because this branch is now pretty long and complicated for a codepath that we'll almost never go down. But maybe that's okay.

> have you tried using many workers?

I haven't tried pangeo-style workloads with >30 workers, but I have tried my standard shuffle-profile with this which prompted 91aee92 which I need to look into a little more.

Reply:
My hope is that we can get this in today or tomorrow.  Is that hope achievable?  If not, do you have a sense for what a reasonably deadline would be?

Reply:
Today.

Reply:
I shall prepare the happy dance.